tasks/apm-host-injection.yml

```diff
@@ -0,0 +1,35 @@
+---
+- name: Check if platform is supported
+  fail:
+    msg: APM Injection is not supported in this platform
+  when: ansible_facts.os_family not in ["Debian", "RedHat", "Rocky", "AlmaLinux"]
+
+- name: Flush handlers
+  meta: flush_handlers
+
+- name: Ensure datadog-agent is running
+  service:
+    name: datadog-agent
+    state: started
+  register: datadog_agent_service
+  until: datadog_agent_service.status.ActiveState == "active"
+  retries: 3
+  delay: 10
+
+- name: Debian Install Tasks
+  include_tasks: pkg-debian/install-apm-host-inject-latest.yml
+  when: ansible_facts.os_family == "Debian"
+
+- name: RedHat Install Tasks
+  include_tasks: pkg-redhat/install-apm-host-inject-latest.yml
+  when: ansible_facts.os_family in ["RedHat", "Rocky", "AlmaLinux"]
+
+# TODO: enable when flag is ready
+#- name: Check if dd-host-install needs to be run
+#  command: dd-host-install --dry-run
+#  register: dd_host_install_cmd_result
+#  ignore_errors: true
+
+- name: Run dd-host-install
```

Comment:
Couple points here:
* Should this be run conditionally only when `datadog_apm_host_injection_enabled` is set to `true`?
* Are we missing the analogous call for `datadog_apm_docker_injection_enabled`?
* Can we make the name more descriptive for users, e.g. `Run APM host injection setup script`?
* Again, this should get `when: not ansible_check_mode`.

Reply:
Ad first point: what I mean is that when we have both host injection and docker injection options implemented, this shouldn't run when only docker injection is specified, right? Does that make sense?

Reply:
When docker injection is enabled, there's a separate command called `dd-host-container-install`. I've updated the PR with docker injection implemented, let me know if it's clear now!